Update toolkit-provider.doc.stories.mdx (#658)
jpmorganchase/uitk: packages/core/stories/toolkit-provider.doc.stories.mdx

- Theme
- Density
- Breakpoints
- AriaAnnouncer

## API

```js static
import { ToolkitProvider } from "@jpmorganchase/uitk-core";
```

## Props

<ArgsTable of={ToolkitProvider} />

## Theme

> You can use the `theme` prop to change the default theme (set to `light`) by passing a theme name or an array of theme names. For more details on Theme please go to <a href="?path=/story/documentation-styles-and-theming-introduction--page">Styles and Theming</a>

Reviewer: ```suggestion
You can use the `theme` prop to change the default theme (set to `uitk-theme`) by passing a theme name. For more details on Theme please go to <a href="?path=/story/documentation-styles-and-theming-introduction--page">Styles and Theming</a>
```

> You can use the `theme` prop to change the default theme (set to `light`) by passing a theme name or an array of theme names. For more details on Theme please go to <a href="?path=/story/documentation-styles-and-theming-introduction--page">Styles and Theming</a>

Reviewer 2: This needs to be updated to match the new API

### useTheme

You can use the `useTheme` hook to access the value which indicates which theme and mode are currently active. The `useTheme` hook returns an object which contains string values of the currently set theme and mode in scope. You can optionally override these values by providing theme name and mode strings to `ToolkitProvider` via the `theme` and `mode` props.

```tsx
import { useTheme } from "@jpmorganchase/uitk-core";

function MyComponent() {
  const { mode, theme } = useTheme();
  return (
    <div>
      <div>Current theme: {theme}</div>
      <div>Current mode: {mode === "light" ? "Light Mode" : "Dark Mode"}</div>
    </div>
  );
}
```

## Density

You can use the `density` prop to change the density on the wrapped component or app, which by default is set to `medium`.

### useDensity

> You can use the `useDensity` hook to access the currently active density of your app in the provided scope. The useDensity hook returns the string value of the currently set density. The possible values are `touch`, `low`, `medium` and `high`. You can optionally override this value by passing your new value as the first parameter to the function.

Reviewer 2: ```suggestion
You can use the `useDensity` hook to access the currently active density of your app in the provided scope. The `useDensity` hook returns the string value of the currently set density. The possible values are `touch`, `low`, `medium` and `high`. You can optionally override this value by passing your new value as the first parameter to the function.
```

}
```

## <a name="breakpoints"></a>Breakpoints

> The `breakpoints` prop on ToolkitProvider can be used to customize the css media queries that custom hooks like `useBreakpoints` and `useCurrentBreakpoint` are using internally. These hooks are usually used to enable the responsive props, where you can pass multiple values (one per breakpoint) or a single value that would take effect across all different screen sizes. These props can mostly be found in the layout components for now. Refer to the [FlexLayout](/docs/documentation-core-layout-flexlayout--page) component documentation for a good example of this.

Reviewer 2: ```suggestion
The `breakpoints` prop on `ToolkitProvider` can be used to customize the CSS media queries that custom hooks like `useBreakpoints` and `useCurrentBreakpoint` are using internally. These hooks are usually used to enable the responsive props, where you can pass multiple values (one per breakpoint) or a single value that would take effect across all different screen sizes. These props can mostly be found in the layout components for now. Refer to the [FlexLayout](/docs/documentation-core-layout-flexlayout--page) component documentation for a good example of this.
```

function MyComponent() {
  const breakpoints = useBreakpoints();
  // here we can use useBreakpoints to access the value from the ToolkitProvider prop
  return <div>Breakpoints: {breakpoints}</div>;
}
```

## AriaAnnouncer

> `AriaAnnouncer` will be injected automatically inside the root element when ToolkitProvider is wrapping the app. If you use

Reviewer 2: ```suggestion
`AriaAnnouncer` will be injected automatically inside the root element when `ToolkitProvider` is wrapping the app. If you use
```

> nested ToolkitProviders, these will not inject further instances of AriaAnnouncer.

Reviewer 2: ```suggestion
nested ToolkitProviders, these will not inject further instances of `AriaAnnouncer`.
```
Have to figure out what we do with `ToolkitProviders` here does it get back ticks or is the s excluded.

Read [AriaAnnouncer documentation](/docs/documentation-core-aria-announcer--page) for more details.

## Examples of ToolkitProvider use

> Here is a `Card` including some very simple content and wrapped with a ToolkitProvider. We set the ToolkitProvider to use the light mode and high density.

Reviewer 2: ```suggestion
Here is a `Card` including some very simple content and wrapped with a `ToolkitProvider`. We set the `ToolkitProvider` to use the light mode and high density.
```

<Canvas>
  <Story name="Simple ToolkitProvider - Touch Density">
    <ToolkitProvider density="high" mode="light">
      <Card>
        <div>
          <h1>This is Card</h1>
          <span>Using Nested DOM Elements</span>
        </div>
      </Card>
    </ToolkitProvider>
  </Story>
</Canvas>

> Here is another example of a `Card` containing an additional nested `Card`. The nested `Card` is also wrapped with a ToolkitProvider, this time configured to use dark mode. Density is not specified on the inner ToolkitProvider, so is inherited from the outer instance.

Reviewer 2: ```suggestion
Here is another example of a `Card` containing an additional nested `Card`. The nested `Card` is also wrapped with a `ToolkitProvider`, this time configured to use dark mode. Density is not specified on the inner ToolkitProvider, so is inherited from the outer instance.
```

Some browser implementations of `display: contents` are buggy (Safari, for example). Further, your code may rely on CSS query selectors in JavaScript that might be broken if they do not anticipate the additional `div` element inserted by the ToolkitProvider.

If you are concerned about either of these scenarios, the `ToolkitProvider` can be configured to not inject this additional element.
Instead it can add the required CSS classes to the child element nested within the Provider. You may also simply prefer this approach.

There is (currently) one limitation to this pattern, you must only use a single child element. This is enforced within the code and, for TypeScript users, via the typings. This would look as follows:

```jsx
<ToolkitProvider mode="light" density="medium" applyClassesTo="child">
  <Card>...</Card>
</ToolkitProvider>
```

Then we would see the following HTML.

```html
<div class="uitkCard uitk-theme uitk-density-medium" data-mode="light">...</div>
```

> The `ToolkitProvider` can also be configured to add the required CSS classes to the root HTML element. This would allow you to apply styles to the whole of the document rather than just the root `div` that React renders to. Some use cases for this would be to prevent overscroll or to apply styling to the whole background of the page.

Reviewer 2: This terminology is a bit confusing: "rather than just the root `div` that React renders to"

Reviewer: How about "rather than just the `div` given to ReactDOM to render to"

Reviewer 2: I wonder if we could just get rid of the sentence. It doesn't add much

```suggestion
The `ToolkitProvider` can also be configured to add the required CSS classes to the root HTML element. This would allow you to apply styles to the whole of the document. Some use cases for this would be to prevent overscroll or to apply styling to the whole background of the page.
```
